components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -1763,13 +1763,21 @@ public void makeAPIKeysForwardCompatible(API api) throws APIManagementException
         String apiName = api.getId().getApiName();
         Set<String> versions = getAPIVersions(provider, apiName);
         APIVersionComparator comparator = new APIVersionComparator();
+        List<API> sortedAPIs = new ArrayList<API>();
         for (String version : versions) {
             API otherApi = getAPI(new APIIdentifier(provider, apiName, version));
-            if (comparator.compare(otherApi, api) < 0 && !(APIConstants.RETIRED.equals(otherApi.getStatus()))) {
-                apiMgtDAO.makeKeysForwardCompatible(provider, apiName, version,
-                        api.getId().getVersion(), api.getContext());
+            if (comparator.compare(otherApi, api) < 0 && !otherApi.getStatus().equals(APIStatus.RETIRED)) {
```

Comment:
otherApi can be null. Please change the order

Reply:
Add a null check to the api retrieved by the version.
https://github.com/wso2/carbon-apimgt/pull/6276/commits/a1b307506800771148d5bdf7d51cfffb59fb08e4